packages/ibm-products/src/components/ConditionBuilder/utils/handleKeyboardEvents.js

```diff
@@ -0,0 +1,150 @@
+import { blockClass } from '../ConditionBuilderContext/DataConfigs';
+import { checkForHoldingKey } from './checkForHoldingKey';
+import { focusThisField, traverseClockVise, traverseReverse } from './util';
+
+export const handleKeyDown = (e, conditionBuilderRef) => {
+  const handleKeyPressForPopover = (e, parentContainer) => {
+    const key = e.key;
+    const isHoldingShiftKey = checkForHoldingKey(e, 'shiftKey');
+    const isMultiSelect =
+      parentContainer.querySelector('ul')?.dataset.multiSelect;
+    let allItems = [];
+    switch (key) {
+      case 'ArrowUp':
+        //traverse through the popover options, search box, selectAll button
+        parentContainer
+          .querySelectorAll(`[role="option"]`)
+          .forEach((eachElem, index, allElements) => {
+            traverseReverse(eachElem, index, allElements);
+          });
+
+        break;
+      case 'ArrowDown':
+        //traverse through the popover options, search box, selectAll button
+        parentContainer
+          .querySelectorAll(`[role="option"]`)
+          .forEach((eachElem, index, allElements) => {
+            traverseClockVise(eachElem, index, allElements);
+          });
+        break;
+
+      case 'Tab':
+        allItems = [
+          ...Array.from(
+            parentContainer.querySelectorAll(
+              `.${blockClass}__selectAll-button,[role="searchbox"]`
+            )
+          ),
+          parentContainer.querySelector(`[role="option"]`),
+        ];
+
+        allItems.forEach((eachElem, index, allElements) => {
+          if (isHoldingShiftKey) {
+            traverseReverse(eachElem, index, allElements, true, true);
+          } else {
+            traverseClockVise(eachElem, index, allElements, true, true);
+          }
+        });
+        e.preventDefault();
+        break;
+
+      case ' ':
+        if (isMultiSelect === 'true') {
+          if (document.activeElement.type !== 'button') {
+            //for button , enter key is click which already handled by framework, for other elements trigger click
+            document.activeElement?.click();
+          }
+          e.preventDefault();
+        }
+
+        break;
+      case 'Enter':
+        if (isMultiSelect !== 'true') {
+          if (document.activeElement.type !== 'button') {
+            //for button , enter key is click which already handled by framework, else trigger click
+            document.activeElement?.click();
+            focusThisField(e);
+          }
+        }
+
+        break;
+      case 'Escape':
+        //focus the corresponding field in which the popover is triggered
+        focusThisField(e);
+        break;
+      default:
+        break;
+    }
+  };
+
+  function getRows() {
```

Comment:
for consistency sake lets keep the arrow function notation

Reply:
Hey,
Modified as advised